find_max.rb

```diff
@@ -0,0 +1,19 @@
+
+# Find the maximum element between index 0 and max_index, exclusive
+
+def find_max(array, max_index)
```

Comment:
I had to read this a few times before understanding what `max_index` meant.  Could we use `length` instead?

```suggestion
def find_max(array, length)
```